reana/cli.py

```diff
@@ -477,9 +490,40 @@ def select_components(components):
         else:
             display_message('Ignoring unknown component {0}.'.format(
                 component))
+
+    if exclude_components:
+        output = exclude_components_from_selection(output, exclude_components)
+
     return list(output)
 
 
+def exclude_components_from_selection(selection, exclude_components):
+    """Exclude list of components from list of selections.
+
+    :param selection: List of selected components in standard naming form.
+    :param exclude_components: List of components to exclude, either in short
+        or standard naming form.
+    :type selection: set
+    :type exclude_components: list
+
+    :return: Set of selected components without ``exclude_components``, all in
+        standard naming form.
+    :rtype: set
+    """
+    st_exclude_components = \
```

Comment:
What is `st`?

Reply:
Clarified.

Reply:
`st` is my terminal :wink: https://st.suckless.org/